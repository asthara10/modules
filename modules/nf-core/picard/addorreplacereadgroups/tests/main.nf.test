nextflow_process {

    name "Test Process PICARD_ADDORREPLACEREADGROUPS"
    script "../main.nf"
    config "./nextflow.config"
    process "PICARD_ADDORREPLACEREADGROUPS"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/addorreplacereadgroups"

    test("test_picard_addorreplacereadgroups") {


        when {
            process {
                """
                
                input[0] = [ 
        [ id:'test', single_end:false ], // meta map
        file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true) 
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

}
