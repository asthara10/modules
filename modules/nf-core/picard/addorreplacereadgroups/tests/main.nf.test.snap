{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d893d88521b29d71b2eee9be6adfc9ef"
            ]
        ],
        "timestamp": "2023-11-24T13:13:39.866104"
    }
}